### PR TITLE
WSO-541 - Change Local Login page

### PR DESCRIPTION
**Pull request by SandyVallee**

### CODE REVIEW

File: sites/all/modules/ucalgary/ucgy_core/ucgy_core_roles/ucgy_core_roles.module [hunk 1]
```diff
@@ -241,6 +241,14 @@ function ucgy_core_roles_form_alter(&$form, &$form_state, $form_id) {
             break;
 
           case 'user/login':
+            // are we literally on /user/login
+           //var_dump($form);
```
Comment (dpfitzsi) [inline sites/all/modules/ucalgary/ucgy_core/ucgy_core_roles/ucgy_core_roles.module]: Get rid of debug

File: sites/all/themes/ucgy_theme/includes/preprocess.inc [hunk 2]
```diff
@@ -3,6 +3,8 @@
 // PREPROCESS VARIABLES FOR html.tpl.php
 function ucgy_theme_preprocess_html(&$variables)
 {
+  var_dump($variables['page']['content']['system_main']['name']['#description']);
```
Comment (dpfitzsi) [inline sites/all/themes/ucgy_theme/includes/preprocess.inc]: Get rid of var_dump.

Are these changes still needed or should this file be reverted?

File: sites/all/themes/ucgy_theme/includes/preprocess.inc [hunk 3]
```diff
@@ -33,6 +35,7 @@ function ucgy_theme_preprocess_html(&$variables)
 // PREPROCESS VARIABLES FOR page.tpl.php
 function ucgy_theme_preprocess_page(&$variables)
 {
+  //var_dump($_SERVER);// also works
```
Comment (dpfitzsi) [inline sites/all/themes/ucgy_theme/includes/preprocess.inc]: Remove

File: sites/all/themes/ucgy_theme/includes/preprocess.inc [hunk 4]
```diff
@@ -59,6 +62,7 @@ function ucgy_theme_preprocess_page(&$variables)
 // PREPROCESS VARIABLES FOR region.tpl.php
 function ucgy_theme_preprocess_region(&$variables)
 {
+  //var_dump($_SERVER);// works
```
Comment (dpfitzsi) [inline sites/all/themes/ucgy_theme/includes/preprocess.inc]: Remove

File: sites/all/themes/ucgy_theme/includes/preprocess.inc [hunk 5]
```diff
@@ -85,6 +90,7 @@ function ucgy_theme_preprocess_menu_tree(&$variables)
 
 function ucgy_theme_preprocess_menu_link(&$variables)
 {
+  //var_dump($_SERVER); // works!
```
Comment (dpfitzsi) [inline sites/all/themes/ucgy_theme/includes/preprocess.inc]: Remove

File: sites/all/themes/ucgy_theme/includes/preprocess.inc [hunk 6]
```diff
@@ -68,6 +72,7 @@ function ucgy_theme_preprocess_region(&$variables)
 // PREPROCESS VARIABLES FOR block.tpl.php
 function ucgy_theme_preprocess_block(&$variables)
 {
+  //var_dump($_SERVER);// works!
```
Comment (dpfitzsi) [inline sites/all/themes/ucgy_theme/includes/preprocess.inc]: Remove

File: sites/all/themes/ucgy_theme/includes/preprocess.inc [hunk 7]
```diff
@@ -100,6 +106,7 @@ function ucgy_theme_preprocess_menu_link(&$variables)
  */
 function ucgy_theme_preprocess(&$variables, $hook)
 {
+ // var_dump($_SERVER);
```
Comment (dpfitzsi) [inline sites/all/themes/ucgy_theme/includes/preprocess.inc]: Remove

File: sites/all/themes/ucgy_theme/includes/preprocess.inc (hunk 2 above)
Comment (SandyVallee) [reply sites/all/themes/ucgy_theme/includes/preprocess.inc]: I'll revert it sorry I missed these, I guess my revert didn't work. I'll try again

File: sites/all/modules/ucalgary/ucgy_core/ucgy_core_roles/ucgy_core_roles.module (hunk 1 above)
Comment (SandyVallee) [reply sites/all/modules/ucalgary/ucgy_core/ucgy_core_roles/ucgy_core_roles.module]: done

File: sites/all/modules/ucalgary/ucgy_core/ucgy_core_roles/ucgy_core_roles.module [hunk 8]
```diff
@@ -241,6 +241,13 @@ function ucgy_core_roles_form_alter(&$form, &$form_state, $form_id) {
             break;
 
           case 'user/login':
+            // are we literally on /user/login
```
Comment (taeram) [inline sites/all/modules/ucalgary/ucgy_core/ucgy_core_roles/ucgy_core_roles.module]: These changes are pretty much the same as what's in the `case 'user':` above, so it would be nice to combine the two, and wrap everything in `if (module_exists('simplesamlphp_auth')) {`, as they both rely on simplesaml being enabled.

Replacing the entire `switch` statement with something like this would be better:
```php
if (module_exists('simplesamlphp_auth')) {
  $saml_loginText = [
    '#markup' => '<br><a href="/saml_login">Please click here to log in with your UCalgary credentials</a><br><br>',
  ];
  array_unshift($form, $saml_loginText);
  $form['#actions'] = '/saml_login';

  if ($_GET['q'] == 'user') {
    hide($form['name']);
    hide($form['pass']);
  }
}
```

Comment (SandyVallee) [reply sites/all/modules/ucalgary/ucgy_core/ucgy_core_roles/ucgy_core_roles.module]: done with a small change to this.
